src/murmur/Messages.cpp

```diff
@@ -77,6 +77,59 @@
 		sendMessage(uSource, mppd); \
 	}
 
+/// A helper class for managing temporary access tokens
+/// It will add the tokens in the comstructor and remove them again in the destructor effectively
+/// turning the tokens into a scope-based property
+class TemporaryAccessTokenHelper {
+	protected:
+		ServerUser *affectedUser;
+		QStringList qslTemporaryTokens;	
+		Server *server;
+
+	public:
+		TemporaryAccessTokenHelper(ServerUser *affectedUser, const QStringList &tokens, Server *server) : affectedUser(affectedUser),
+			qslTemporaryTokens(tokens), server(server) {
+			// Add the temporary tokens
+			QMutableStringListIterator it(this->qslTemporaryTokens);
+
+			{
+				QMutexLocker qml(&server->qmCache);
```

Comment:
Do I really have to aquire this lock for accessing the token lost of a user? From the name I would assume I have to use it when accessing the cache instead...

Reply:
@davidebeatrici do you know an answer to that? I think this is the only thing holding up a merge :)

Reply:
Unfortunately not, we would have to check whether the new code accesses something that may be changed by another thread.

Reply:
Okay. I took this from
https://github.com/mumble-voip/mumble/blob/281fe9e5f38e7d1ba4610b03810cc944ab2de5a9/src/murmur/Messages.cpp#L85-L89
There at least this lock is also used for accessing ` qslAccessTokens` of a user, but not for actually changing the cache, which I think is a bit weird, but I guess if you don't know either, I'll stick to this pattern as well...